### DISCUSSION TITLE
[Queue] Set retry_after for specific queues or jobs

### DISCUSSION THREAD

**Discussion by philipkozeny** [laravel/framework#33661] [Ideas]
Currently, there is a global retry_after setting for a queue connection (https://laravel.com/docs/7.x/queues#job-expirations-and-timeouts). 

In our case, we have different queues within one connection which span from super quick jobs (like notifications) to jobs that can take up to twelve hours (data generation). Setting the retry_after to the longest-running queue would be problematic for time-sensitive jobs that should be picked up again rather quickly if something happens to the worker. On the other hand having multiple workers starting the same job again since its available again although the process is still running is also not optimal.

It would be great to either be able to
a) assign a specific retry_after for a queue 
b) assign a specific retry_after for a job itself when dispatching it

**Comment by ChrisSantiago82** [comment]
I have pretty much the same question.
Inside a job, you can define a timeout for that specific job. For example, if you have long running jobs or something like this.
But then there is one global setting got retry_after. And it sais the retry_after should be bigger than the timeout.
What is the best way to handle this?

https://laravel.com/docs/8.x/queues#timeout
